### ISSUE TITLE
UglifyJS errors don't bail the build with --bail

### ISSUE THREAD

**Issue by graingert** [webpack/webpack#2390]
webpack --bail:

``` js
ERROR in entrypoint-xxx.bundle.js from UglifyJs
Unexpected character '`' ./src/path/to/file.js:1487,0]
```

**Comment by bebraw** [comment]
Can you provide a test case so this can be debugged?

**Comment by pdillon** [comment]
@bebraw Check out at https://github.com/pdillon/webpack-bail-example. Ran into this issue where the babel loader was pulling in files outside of the include path but Uglify bombs out.

**Comment by bebraw** [comment]
@pdillon Thanks. I added this to webpack 2 work. Given there's a test case, it should be easier to figure out the root cause.

**Comment by pdillon** [comment]
@bebraw would it make sense for bail to map to the NoErrorsPlugin, and for the plugin to pass an error to the callback if compilation.errors > 0?

**Comment by bebraw** [comment]
@pdillon I'm not sure. I think we'll have to wait for Tobias to check this out to get the behavior right.

**Comment by gdborton** [comment]
Is this not going to be fixed in webpack 1.13.x?

**Comment by q42jaap** [comment]
I've created a workaround that will detect UglifyJs errors and exit the process with a non-zero status code:
```
class DetectUglifyJsErrors {
  apply(compiler) {
    compiler.plugin('compilation', compilation => {
      compilation.plugin('optimize-chunk-assets', (chunks, callback) => {
        if (compilation.errors.length > 0) {
          compilation.errors.forEach(error => {
            console.error(`ERROR in ${error.message}`);
          });
          process.exit(1);
        }
        callback();
      });
    });
  }
}
```
This works for me and outputs the same message webpack would otherwise output.
